packages/round-manager/src/features/api/program.ts

```diff
@@ -34,34 +33,31 @@ export async function listPrograms(
       throw Error("Unable to fetch programs");
     }
 
-    let programs: Program[] = [];
+    const programs: Program[] = [];
 
     for (const program of programsRes.programs) {
-      programs.push({
-        id: program.id,
-        metadata: program.metadata,
-        operatorWallets: program.roles.map(
-          (role: { address: string }) => role.address
-        ),
-        tags: program.tags,
-        chain: {
-          id: chainId,
-          name: CHAINS[chainId]?.name,
-          logo: CHAINS[chainId]?.logo,
-        },
-        createdByAddress: program.createdByAddress,
-      });
+      if (
+        program.roles.some(
+          (role) => role.role === "OWNER" && role.address === address
```

Comment:
just to confirm do we have other roles other than OWNER? like MEMBER? and do they also need access?

Reply:
i am assuming a bot cannot fake the creation of a profile with someone else being an owner?

Reply:
> just to confirm do we have other roles other than OWNER? like MEMBER? and do they also need access?

we also have members, but this the original filter was filtering by `createdByAddress` I assume that we only show the Program to the actual Owner/Operator

@meglister I think that could be a product question: Should the Program also be shown to Program Member?
Problem: A Member of a Program would see all the rounds, but since Program Member **are not always** round manager, the members would also see rounds, which they may can't access.

My Suggestion: keep the owner for now and start a discussion about the additional flows we would need

Reply:
> My Suggestion: keep the owner for now and start a discussion about the additional flows we would need

This makes sense. Let's wait for meg to sign off and then move ahead

Reply:
the filter we have was for a problem we had in Scroll. bots where creating projects/programs adding other people just to spam. 
so I would end up seeing fake programs where I'm manager, but they are actually spam programs. 
It happened only in Scroll, I wonder if it's still happening, we can try and maybe remove the filter if it's no longer the case. 

if it's no longer needed, we need to go back to load all the programs where I have any role, both owner and member, and we don't need to filter anything anymore in the front-end.

Now it filters out projects if the sender is not part of the managers, because bots were creating projects replying data from other tx, without adding themselves to the members.

Reply:
> i am assuming a bot cannot fake the creation of a profile with someone else being an owner?

@boudra  they can, in fact that's the temporary filter we use to understand if they were the bots we had in scroll

Reply:
@gravityblast yeah that's what i thought, thanks for the confirmation, the check if the user is part of the roles is already done in the graphql in this case. the check here is for the spam in scroll.

we should check scroll, i remember the bots were just repeating any transaction we made, i wonder if there is a way to the check but make it work for multisig?

Reply:
note that if we do decide to refactor how we do the check here, please let's keep the following comment (or a variation of it), because it's useful context.

```
    // Filter out programs where operatorWallets does not include round.createdByAddress.
    // This is to filter out spam rounds created by bots
```

Reply:
if we do find out that it's impossible to do the bot spam check with multisigs, maybe we can solve it at the product level and introduce a hide / archive feature for programs & rounds. so users can get rid of spam rounds.

Reply:
we can try!
the spam problem is more for Manager than Explorer though. with those bots what happened was:

* I create a round called X
* bots create 5 rounds called X adding me as operator
* when I open Manager I see 6 rounds called X